eciYali/scripts/changeMedia_batch.m

```diff
@@ -0,0 +1,115 @@
+function [model,pos] = changeMedia_batch(model,c_source,media,flux)
+% changeMedia_batch
+%
+% function that modifies the ecModel and makes it suitable for batch growth
+% simulations on different carbon sources.
+%
+% model:  An enzyme constrained model
+% meadia: Media type ('YEP' for complex, 'MAA' minimal with Aminoacids,
+%                          'Min' for minimal media)
+% flux:   (Optional) A cell array with measured uptake fluxes in mmol/gDwh
+%
+% model: a constrained ecModel
+%
+% usage: [model,pos] = changeMedia_batch(model,c_source,media,flux)
+%
+% Ivan Domenzain        2019-05-23
+
+% Give the carbon source (c_source) input variable with the following
+% format: c_source  = 'D-glucose exchange (reversible)'
+
+if nargin<3
+    media = 'Min';
+end
+%first block any uptake
+[rxnIDs,exchange]  = getExchangeRxns(model);
+%Exclude protein pool from exchange reactions list
+protIndex = find(contains(model.rxnNames,'prot_'));
+exchange  = setdiff(exchange,protIndex);
+%First allow any exchange (uptakes and secretions)
+model.ub(exchange) = Inf;
+%Then block all uptakes
+uptakes            = exchange(find(contains(rxnIDs,'_REV')));
+model.ub(uptakes)  = 0;
+pos = getComponentIndexes(model,c_source);
+
+%Block O2 and glucose production (avoids multiple solutions):
+model.ub(strcmp(model.rxnNames,'oxygen exchange'))    = 0;
+model.ub(strcmp(model.rxnNames,'D-glucose exchange')) = 0;
+%Find substrate production rxn and block it:
+pos_rev = strcmpi(model.rxnNames,c_source(1:strfind(c_source,' (reversible)')-1));
+model.ub(pos_rev) = 0;
+
+%The media will define which rxns to fix:
+if strcmpi(media,'YEP')
+    N = 25;     %Aminoacids + Nucleotides
+elseif strcmpi(media,'MAA')
+    N = 21;     %Aminoacids
+elseif strcmpi(media,'Min')
+    N = 1;      %Only the carbon source
+end
+%UB parameter (manually optimized for glucose on Min+AA):
+b = 0.08;
+%UB parameter (manually optimized for glucose complex media):
+c = 2;
+%Define fluxes in case of ec model:
+if nargin < 4   %Limited protein    
+    if N>1
+       flux    = b*ones(1,N);
+       if N>21
+           flux(22:25) = c;
+       end
+    end
+    flux(1) = Inf;
```

Comment:
Why leave glucose completely unconstrained? This creates the problem we discussed in https://github.com/SysBioChalmers/ecModels/pull/18#issuecomment-542223407: `eciYali` grows unrestricted, where instead we would expect it to grow at the same growth rate of `iYali`. I think unconstraining glucose should only be done in `eciYali_batch`, let me know what you think.